contracts/swap/contract/code.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Package swap package wraps the 'swap' Ethereum smart contract.
+// It is an abstraction layer to hide implementation details about the different
+// Swap contract iterations (SimpleSwap, Swap, etc.)
+package contract
+
+// ContractDeployedCode is used to detect suicides. This constant needs to be
+// updated when the contract code is changed.
+// **CURRENT: SIMPLESWAP**
+const ContractDeployedCode = ""
```

Comment:
This same `const` is defined elsewhere (`contracts/chequebook/contract/code.go`) with a non empty value. Can you please explain this in a bit more detail? Thanks

Reply:
As explained in the PR description, everything in `contracts/chequebook` has to be considered legacy and should be removed. It is there only for reference until we are confident that we did not forget anything relevant from that part.

I am not sure we need this value at all. It is empty now because I don't know exactly if we need it and what should go there. I understand it to hold the **last** contract address deployed and being used, so that we don't accidentally interact with an old defunct (suicided) contract (which would mean losing funds).

Check with @eknir and @ralph-pichler and possibly also with @zelig if we need this feature and/or we can design it in a different way. I think this may be an old design pattern.